### PR TITLE
Added signature creation / validation and removed signed envelope

### PR DESCRIPTION
**Pull request by evik42**
Removed signed envelope and made certificate and node descriptor structure more straight forward by including the signatures.
Added fingerprint and keypair generation, signing and verifying signature.

### CODE REVIEW

File: src/cryptography.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+fn sign_bytes(bytes: impl AsRef<[u8]>, secret_key: &SecretKey) -> Vec<u8> {
+    let expanded_secret_key = ExpandedSecretKey::from(secret_key);
+    let public_key = PublicKey::from(secret_key);
+    let signature_value = expanded_secret_key.sign(bytes.as_ref(), &public_key);
+    signature_value.to_bytes().into()
+}
+
+pub fn verify_signature_json(
```
Comment (MrDarthShoe) [inline src/cryptography.rs:114]: Wouldn't this function require also `EncryptionAlgorithm` when new one is added?

File: src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+use std::fs::File;
+use std::io::{BufWriter, Write};
+use std::path::Path;
+
+use anyhow::Result;
+use serde::Serialize;
+
+pub fn save_json_to_file<C: ?Sized + Serialize>(path: impl AsRef<Path>, content: &C) -> Result<()> {
```
Comment (MrDarthShoe) [inline src/utils.rs:8]: This module is not used. Please add it later when needed / mark it as public in main `lib.rs` / mark it as `#[allow(dead_code)]`

File: src/serde_jcs/mod.rs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+mod entry;
+mod serializer;
+
+pub use self::serializer::to_string;
+pub use self::serializer::to_vec;
```
Comment (MrDarthShoe) [inline src/serde_jcs/mod.rs:5]: Consider making whole `serde_jcs` a separate lib crate in this repo as it cargo shows a warning that some parts are not used. Or mark `#[allow(dead_code)]` temporarily

File: src/cryptography.rs (hunk 1 above)
Comment (evik42) [reply src/cryptography.rs:114]: Both sign and verify would need a hash, the encryption algorithm is included with the Key. But for now it is going with the default.

**Review by MrDarthShoe — APPROVED** [review]

**Comment by MrDarthShoe** [comment]
Part of https://github.com/golemfactory/yagna/issues/2443

**Pull request merged**